packages/eslint-plugin/src/rules/emotion-to-compiled/index.tsx

```diff
@@ -0,0 +1,95 @@
+import type { Rule } from 'eslint';
+
+import type { ImportSpecifier, ImportDeclaration } from 'estree';
+
+const hasStyledImport = (node: ImportDeclaration) => node.source.value === '@emotion/styled';
+const hasCoreImport = (node: ImportDeclaration) => node.source.value === '@emotion/core';
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      recommended: true,
+    },
+    fixable: 'code',
+    type: 'problem',
+    messages: {
+      noStyled: `The '@emotion/styled' library should no longer be in use. Use '@compiled/react'.`,
+      noCore: `The '@emotion/core' library should no longer be in use. Use '@compiled/react'.`,
+      noPragma: `The /** @jsx jsx */ pragma is not require in '@compiled/react'. It can be safely removed.`,
+    },
+  },
+  create(context) {
+    return {
+      Program() {
+        const pragma = context
+          .getSourceCode()
+          .getAllComments()
+          .find((n) => n.value === '* @jsx jsx ');
+        if (pragma) {
+          return context.report({
+            messageId: 'noPragma',
+            loc: pragma.loc!,
+            fix(fixer) {
+              return fixer.replaceText(pragma as any, "import * as React from 'react';");
+            },
+          });
+        }
+      },
+      ImportDeclaration(node) {
+        const hasStyled = hasStyledImport(node);
```

Comment:
Can you place these before the `if` statements, so there's slightly less work to be done depending on when the check succeeds

Reply:
sure.